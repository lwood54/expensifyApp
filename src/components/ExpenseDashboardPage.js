import React from 'react';
import ExpenseList from './ExpenseList';
import ExpenListFilters from './ExpenseListFilters';

const ExpenseDashboardPage = () => (
  <div>
    <ExpenListFilters />
    <ExpenseList />
  </div>
);

export default ExpenseDashboardPage;
