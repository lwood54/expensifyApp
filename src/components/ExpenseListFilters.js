import React from 'react';
import { connect } from 'react-redux';
import { setTextFilter } from '../actions/filters';

const ExpenseListFilters = props => (
  <div>
    <input
      type="text"
      value={props.filters.text}
      onChange={ev => {
        props.dispatch(setTextFilter(ev.target.value));
      }}
    />
  </div>
);

const mapStateToProps = state => {
  return {
    filters: state.filters
  };
};

export default connect(mapStateToProps)(ExpenseListFilters);
