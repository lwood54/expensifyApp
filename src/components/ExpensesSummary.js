import React from 'react';
import { connect } from 'react-redux';
import selectExpenses from '../selectors/expenses';
import getExpensesTotal from '../selectors/expenses-total';
import numeral from 'numeral';

export const ExpensesSummary = props => {
  const expenseWord = props.expenses.length === 1 ? 'expense' : 'expenses';
  const isVare = props.expenses.length === 1 ? 'is' : 'are';
  return (
    <div>
      <h4>
        There {isVare} {props.expenses.length} {expenseWord}.
      </h4>
      <h5>
        Total {expenseWord}:{' '}
        {numeral(props.expensesTotal / 100).format('$0,0.00')}
      </h5>
    </div>
  );
};

const mapStateToProps = state => ({
  expenses: selectExpenses(state.expenses, state.filters),
  expensesTotal: getExpensesTotal(selectExpenses(state.expenses, state.filters))
});

export default connect(mapStateToProps)(ExpensesSummary);
