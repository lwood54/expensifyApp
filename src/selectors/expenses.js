import moment from 'moment';

// GET VISIBLE EXPENSES
const getVisibleExpenses = (expenses, { text, sortBy, startDate, endDate }) => {
  return expenses
    .filter(expense => {
      const createdAtMoment = moment(expense.createdAt);
      const startDateMatch = startDate
        ? startDate.isSameOrBefore(createdAtMoment, 'day')
        : true;
      const endDateMatch = endDate
        ? endDate.isSameOrAfter(createdAtMoment, 'day')
        : true;
      const textMatch = expense.description
        .toLowerCase()
        .includes(text.toLowerCase());
      return startDateMatch && endDateMatch && textMatch;
    })
    .sort((a, b) => {
      if (sortBy === 'date') {
        return a.createdAt < b.createdAt ? 1 : -1; // newest to oldest
      } else if (sortBy === 'amount') {
        return a.amount < b.amount ? 1 : -1; // highest to lowest
      }
    });
};

export default getVisibleExpenses;
