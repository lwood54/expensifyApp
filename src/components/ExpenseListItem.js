import React from 'react';
import { connect } from 'react-redux';
import { removeExpense } from '../actions/expenses';

const ExpenseListItem = ({ description, amount, createdAt, id, dispatch }) => (
  <div>
    <li>
      <h3>{description}</h3>
      <p>
        {amount}: {createdAt}
      </p>
      <button
        onClick={() => {
          dispatch(removeExpense({ id }));
        }}
      >
        Remove
      </button>
    </li>
  </div>
);

// const ExpenseListItem = props => (
//   <div>
//     <li>
//       <h3>{props.description}</h3>
//       <p>
//         {props.amount}: {props.createdAt}
//       </p>
//       <button
//         onClick={() => {
//           props.dispatch(removeExpense({ id: props.id }));
//         }}
//       >
//         Remove
//       </button>
//     </li>
//   </div>
// );

export default connect()(ExpenseListItem);
