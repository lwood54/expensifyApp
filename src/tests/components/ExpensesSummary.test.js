import React from 'react';
import { ExpensesSummary } from '../../components/ExpensesSummary';
import expenses from '../fixtures/expenses';
import { shallow } from 'enzyme';

test('should render 1 expense with its total', () => {
  const wrapper = shallow(
    <ExpensesSummary expenses={[expenses[1]]} expensesTotal={235} />
  );
  expect(wrapper).toMatchSnapshot();
});

test('should render 2 expenses with their total', () => {
  const wrapper = shallow(
    <ExpensesSummary
      expenses={[expenses[1], expenses[2]]}
      expensesTotal={450}
    />
  );
  expect(wrapper).toMatchSnapshot();
});
