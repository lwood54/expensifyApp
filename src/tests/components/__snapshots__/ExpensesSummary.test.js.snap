// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1 expense with its total 1`] = `
<div>
  <h4>
    There 
    is
     
    1
     
    expense
    .
  </h4>
  <h5>
    Total 
    expense
    :
     
    $2.35
  </h5>
</div>
`;

exports[`should render 2 expenses with their total 1`] = `
<div>
  <h4>
    There 
    are
     
    2
     
    expenses
    .
  </h4>
  <h5>
    Total 
    expenses
    :
     
    $4.50
  </h5>
</div>
`;
